scope.go

```diff
@@ -197,6 +194,20 @@ func (s *scope) incQueued() error {
 	}
 }
 
+func (s *scope) calculateQueueDeadlineAndSleep() (time.Duration, time.Time) {
+	d := s.maxQueueTime()
+	dSleep := d / 10
+	if dSleep > time.Second {
+		dSleep = time.Second
+	}
+	if dSleep < time.Millisecond {
+		dSleep = time.Millisecond
+	}
+	deadline := time.Now().Add(d)
+
+	return dSleep, deadline
+}
+
 func (s *scope) inc() error {
 	uQueries := s.user.queryCounter.inc()
 	cQueries := s.clusterUser.queryCounter.inc()
```

Comment:
it feels weird that the LoC 204 to 213 from the original code were not extracted into a function like `checkConcurrency`

Reply:
If I remember correctly, my reasoning was related to the fact that the error variable is shadowed and reused and it was hard to replicate that easily if I changed it into a function.